### PR TITLE
feat: Port all changes from #101

### PR DESCRIPTION
**Pull request by DashKash54**
# What

Bring all the changes from PR #101

### CODE REVIEW

**Review by MaximusHaximus — CHANGES_REQUESTED** [review]
This LGTM! I just had one question about the git submodule still being defined and a nit about clearer error handling on invalid input :)

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule "packages/libs/contracts-sdk/lib/forge-std"]
 	path = packages/libs/contracts-sdk/lib/forge-std
 	url = https://github.com/foundry-rs/forge-std
+[submodule "packages/libs/contracts-sdk/lib/openzeppelin-contracts"]
```
[the remark this answers is not in the document]
Comment (DashKash54) [reply .gitmodules]: I've removed gitmodules since it was messing with nx bundling. But Forge-std can't be added to the package.json unlike @openzepplin/contracts so I've added it in the lib directory: https://github.com/Cyfrin/foundry-full-course-cu/discussions/2084

File: packages/libs/contracts-sdk/test/facets/VincentAppFacet.t.sol [hunk 2]
```diff
@@ -0,0 +1,706 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.29;
```
[the remark this answers is not in the document]
Comment (DashKash54) [reply packages/libs/contracts-sdk/test/facets/VincentAppFacet.t.sol:2]: We have all the contracts edge cases tested in Solidity and the integration test for the ContractsSDK with Jest.

File: packages/libs/contracts-sdk/contracts/facets/VincentUserFacet.sol [hunk 3]
```diff
@@ -44,28 +44,31 @@ contract VincentUserFacet is VincentBase {
...
         string[][] calldata policyIpfsCids,
-        string[][][] calldata policyParameterNames,
-        bytes[][][] calldata policyParameterValues
-    ) external appNotDeleted(appId) onlyPkpOwner(pkpTokenId) onlyRegisteredAppVersion(appId, appVersion) {
+        bytes[][] calldata policyParameterValues
+    ) external appNotDeleted(appId) onlyRegisteredAppVersion(appId, appVersion) appEnabled(appId, appVersion) onlyPkpOwner(pkpTokenId) {
+        uint256 toolCount = toolIpfsCids.length;
+
+        if (toolCount == 0 || policyIpfsCids.length == 0 || policyParameterValues.length == 0) {
+            revert LibVincentUserFacet.InvalidInput();
```
[the remark this answers is not in the document]
Comment (MaximusHaximus) [reply packages/libs/contracts-sdk/contracts/facets/VincentUserFacet.sol:67]: An explicit error message is what i'm referring to preferring over a generic `InvalidInput()` error that has no context about _which input was invalid or why_

**Review by MaximusHaximus — APPROVED** [review]
:+1: LGTM!

File: packages/libs/contracts-sdk/contracts/facets/VincentUserFacet.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (DashKash54) [reply packages/libs/contracts-sdk/contracts/facets/VincentUserFacet.sol:67]: Agreed I can raise a follow-up PR for it
